### PR TITLE
Add support for mixed-precision training

### PR DESCRIPTION
**Pull request by danieldk**
This change makes it possible to use and configure the support for mixed-precision training that was added to thinc.

Example configuration:

```
[components.transformer.model]
@architectures = "spacy-transformers.TransformerModel.v3"
name = "roberta-base"
mixed_precision = true
```

### CODE REVIEW

**Comment by svlandeg** [comment]
I'm afraid we won't be able to merge this PR until a new version of spaCy with Thinc>=8.0.9 is released.

**Comment by svlandeg** [comment]
@danieldk: With spaCy 3.1.3 out, you can go ahead and bump the required spaCy lower pin and take this out of draft :-)

**Comment by danieldk** [comment]
Ready for review 🎉

File: spacy_transformers/architectures.py [hunk 1]
```diff
@@ -130,3 +130,49 @@ def create_TransformerModel_v2(
...
+    """Pretrained transformer model that can be finetuned for downstream tasks.
+
+    name (str): Name of the pretrained Huggingface model to use.
+    get_spans (Callable[[List[Doc]], List[List[Span]]]): A function to extract
+        spans from the batch of Doc objects. See the "TransformerModel" layer
+        for details.
+    tokenizer_config (dict): Settings to pass to the transformers tokenizer.
+    transformers_config (dict): Settings to pass to the transformers forward pass
+        of the transformer.
+    mixed_precision (bool): enable mixed-precision. Mixed-precision replaces
```
Comment (svlandeg) [inline spacy_transformers/architectures.py:153]:
```suggestion
    mixed_precision (bool): Enable mixed-precision. Mixed-precision replaces
```
(style nitpicking)

File: spacy_transformers/architectures.py [hunk 2]
```diff
@@ -130,3 +130,49 @@ def create_TransformerModel_v2(
...
+        whitelisted ops to half-precision counterparts. This speeds up training
+        and prediction on modern GPUs and reduces GPU memory use.
+    grad_scaler_config (dict): configuration for gradient scaling in mixed-precision
```
Comment (svlandeg) [inline spacy_transformers/architectures.py:156]:
```suggestion
    grad_scaler_config (dict): Configuration for gradient scaling in mixed-precision
```

**Review by svlandeg — APPROVED** [review]
Looks good to me!

**Comment by svlandeg** [comment]
You could also add the setting to the `cfg_string` in `test_pipeline_component.py`, then it's tested at least indirectly...

**Review by svlandeg — COMMENTED** [review]
Actually, also the `DEFAULT_CONFIG_STR` in `pipeline_component.py` can be updated.

**Comment by danieldk** [comment]
I just realized that we need a new version of `Tok2VecTransformer` as well.

**Comment by danieldk** [comment]
> Actually, also the `DEFAULT_CONFIG_STR` in `pipeline_component.py` can be updated.

This would be enough, right? Since the transformer in the test uses that as a base configuration and `cfg_string` overrides some settings.

I have now also added `Tok2VecTransformer.v3`.

**Comment by svlandeg** [comment]
Ah, yes, good point! Fiddly listener stuff.

**Pull request merged**